### ISSUE TITLE
can not work well with 15.2.5 

### ISSUE THREAD

**Issue by Damien9527** [Crapworks/ceph-dash#63]
I have install ceph 15.2.5 in centos7  by ceph-deploy.  all goes well , 

# ceph -s
  cluster:
    id:     7a9873c4-60b8-4481-b44b-c29154c406b0
    health: HEALTH_WARN
            Module 'restful' has failed dependency: No module named 'pecan'

  services:
    mon: 3 daemons, quorum ceph01,ceph02,ceph03 (age 19h)
    mgr: ceph03(active, since 19h), standbys: ceph01, ceph02
    mds: cephfs:1 {0=ceph01=up:active} 2 up:standby
    osd: 3 osds: 3 up (since 19h), 3 in (since 10d)

  task status:
    scrub status:
        mds.ceph01: idle

  data:
    pools:   5 pools, 129 pgs
    objects: 29 objects, 693 KiB
    usage:   3.2 GiB used, 22 TiB / 22 TiB avail
    pgs:     129 active+clean

but enable dashboard failed。
 [root@ceph01 ~]# ceph mgr module enable dashboard
Error ENOENT: all mgr daemons do not support module 'dashboard', pass --force to force enablement

so i try to  use ceph-dash .

I run ceph-dash with docker : 
[root@ceph01 ~]# docker run -d -p 5000:5000 -e CEPHMONS='10.188.32.12,10.188.32.13,10.188.32.14' -e KEYRING="$(sudo cat /etc/ceph/ceph.client.admin.keyring)" crapworks/ceph-dash:latest

but in the browers,I got err:  

`KeyError
KeyError: 'osdmap'

Traceback (most recent call last)
File "/usr/lib64/python2.7/site-packages/flask/app.py", line 2464, in __call__
return self.wsgi_app(environ, start_response)
File "/cephdash/app/__init__.py", line 70, in __call__
return self.app(environ, start_response)
File "/usr/lib64/python2.7/site-packages/flask/app.py", line 2450, in wsgi_app
response = self.handle_exception(e)
File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1867, in handle_exception
reraise(exc_type, exc_value, tb)
File "/usr/lib64/python2.7/site-packages/flask/app.py", line 2447, in wsgi_app
response = self.full_dispatch_request()
File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1952, in full_dispatch_request
rv = self.handle_user_exception(e)
File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1821, in handle_user_exception
reraise(exc_type, exc_value, tb)
File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1950, in full_dispatch_request
rv = self.dispatch_request()
File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1936, in dispatch_request
return self.view_functions[rule.endpoint](**req.view_args)
File "/usr/lib64/python2.7/site-packages/flask/views.py", line 89, in view
return self.dispatch_request(*args, **kwargs)
File "/usr/lib64/python2.7/site-packages/flask/views.py", line 163, in dispatch_request
return meth(*args, **kwargs)
File "/cephdash/app/dashboard/views.py", line 118, in get
total_osds = cluster_status['osdmap']['osdmap']['num_osds']
KeyError: 'osdmap'
The debugger caught an exception in your WSGI application. You can now look at the traceback which led to the error.
To switch between the interactive traceback and the plaintext one, you can click on the "Traceback" headline. From the text traceback you can also create a paste of it. For code execution mouse-over the frame you want to debug and click on the console icon on the right side.

You can execute arbitrary Python code in the stack frames and there are some extra helpers available for introspection:

dump() shows all variables in the frame
dump(obj) dumps all that's known about the object
`

don't know where is the problem.   

Need you help.

**Comment by abonanni** [comment]
+1

**Comment by k0ste** [comment]
Can you attach output of `ceph status --format=json-pretty`? Thanks

**Comment by abonanni** [comment]
You can download the file here : https://cloud.teicee.fr/s/oS7fAkC24aEpjs8

**Comment by k0ste** [comment]
Currently is not 'pretty'. Please make file like this: `ceph status --format=json-pretty > ceph.json`, then attach directly to this issue, without cloud hosting. Thanks

**Comment by abonanni** [comment]
Hello.

I can't attach a json file in github (error msg: We don't support that file type).

In the cloud hosting, you have a download button on top :
![Capture d’écran 2021-04-21 à 13 45 18](https://user-images.githubusercontent.com/38793840/115390662-cbdcf800-a1de-11eb-9ade-dc6dbf7b55ad.png)

**Comment by k0ste** [comment]
You just can paste raw, I was use online formatter for pretty source file

```json
{
   "fsid":"24b8843d-92bb-499f-b659-f40d317e920c",
   "health":{
      "status":"HEALTH_WARN",
      "checks":{
         "POOL_TOO_MANY_PGS":{
            "severity":"HEALTH_WARN",
            "summary":{
               "message":"1 pools have too many placement groups",
               "count":1
            },
            "muted":false
         }
      },
      "mutes":[
         
      ]
   },
   "election_epoch":3696,
   "quorum":[
      0,
      1,
      2
   ],
   "quorum_names":[
      "pve4-cae",
      "pve5-cae",
      "pve6-cae"
   ],
   "quorum_age":342612,
   "monmap":{
      "epoch":3,
      "min_mon_release_name":"octopus",
      "num_mons":3
   },
   "osdmap":{
      "epoch":287,
      "num_osds":12,
      "num_up_osds":12,
      "osd_up_since":1618528841,
      "num_in_osds":12,
      "osd_in_since":1618489832,
      "num_remapped_pgs":0
   },
   "pgmap":{
      "pgs_by_state":[
         {
            "state_name":"active+clean",
            "count":513
         }
      ],
      "num_pgs":513,
      "num_pools":2,
      "num_objects":19552,
      "data_bytes":79067914483,
      "bytes_used":787732070400,
      "bytes_avail":5149187702784,
      "bytes_total":5936919773184,
      "read_bytes_sec":0,
      "write_bytes_sec":10995,
      "read_op_per_sec":0,
      "write_op_per_sec":1
   },
   "fsmap":{
      "epoch":1,
      "by_rank":[
         
      ],
      "up:standby":0
   },
   "mgrmap":{
      "available":true,
      "num_standbys":2,
      "modules":[
         "iostat",
         "restful"
      ],
      "services":{
         
      }
   },
   "servicemap":{
      "epoch":7941,
      "modified":"2021-04-20T00:21:43.765089+0200",
      "services":{
         
      }
   },
   "progress_events":{
      
   }
}
```

**Comment by k0ste** [comment]
Seems octopus drop `osdmap` key in `osdmap` collection.

**Comment by abonanni** [comment]
We have maid a PR for ceph 15 : https://github.com/Crapworks/ceph-dash/pull/64

**Comment by Crapworks** [comment]
This should be fixed now by #64. I've created release v1.6 to contain the fix (updated docker containers are now available as well)
